lesson_9/pop_victor/tema9-OOP1-VictorPOP.py

```diff
@@ -0,0 +1,86 @@
+'''
+Build two classes of your choice that can model a real-life example.
+The class needs to meet the following requirements:
+
+at least 5 attributes each
+at least 2 methods each
+one class to inherit from another
+As a demonstration create at least 5 instances of one class (preferably the child class)
+ and call all the methods it holds
+'''
+from datetime import datetime
+
+
+class User(object):
+
+    def __init__(self, nickname, real_name, age, country, join_year):
+        self.nickname = nickname
+        self.real_name = real_name
+        self.age = age
+        self.country = country
+        self.join_year = join_year
+
+    def __repr__(self):
+        return f'Nick: {self.nickname} age: {self.age} from: {self.country}'
+
+    def __str__(self):
+        return f'{self.nickname} joined in {self.join_year}'
+
+    def account_age(self):
+        return datetime.now().year-self.join_year
+
+    def account_restriction(self):
+        company_country = ['Romania', 'Luxembourg', 'Austria', 'Germany']
+        if self.country in company_country:
+            return f'{self.nickname} has access to {self.country} Server'
+        return f'no server in {self.country} at the moment, sorry {self.nickname}'
+
+
+class Gamer(User):
+    def __init__(self, nickname, real_name, age, country, join_year, *args):
+        self.games = args
```

Comment:
Nice use of *ags. An alternative can be to ask for a list and set it as `[]` if the attribute is not specified:

```
def __init__(self, nickname, real_name, age, country, join_year, games=None)
    if games is None:
        self.games = []
    else:
         self.games = games
    super().__init__(nickname, real_name, age, country, join_year)
```

Reply:
the use case was that you have a list of gamers.
thanks for the suggestion I will modify local but I would like to leave it as it is on git.